### PR TITLE
fix: [CI-6659]: populated link in bitbucket onprem

### PR DESCRIPTION
**Pull request by devkimittal**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42944)

### CODE REVIEW

File: product/ci/scm/git/git.go [hunk 1]
```diff
@@ -267,6 +268,13 @@ func GetLatestCommit(ctx context.Context, request *pb.GetLatestCommitRequest, lo
 	}
 
 	refResponse, response, err := client.Git.FindCommit(ctx, request.GetSlug(), ref)
+
+	// bitbucket onprem API doesn't return commit link, hence populating it manually.
+	if refResponse.Link == "" && request.GetProvider().GetBitbucketServer() != nil{
```
Comment (raghavharness) [inline product/ci/scm/git/git.go:273]: 1.  move this after the err check. 
2. Do go fmt of the file.
3. Add a UT under git_test.go.

File: product/ci/scm/git/git.go [hunk 2]
```diff
@@ -267,6 +268,13 @@ func GetLatestCommit(ctx context.Context, request *pb.GetLatestCommitRequest, lo
...
+	    namespace, name := scm.Split(request.GetSlug())
+	    refResponse.Link= fmt.Sprintf("%sprojects/%s/repos/%s/commits/%s", client.BaseURL, namespace, name, refResponse.Sha)
```
Comment (vistaarjuneja) [inline product/ci/scm/git/git.go:275]: we should make sure the assumptions on the base URL are correct (eg - that it ends with `/` here)

**Review by abhinav-harness — APPROVED** [review]

**Comment by devkimittal** [comment]
trigger codeformat

**Comment by devkimittal** [comment]
trigger messagecheck

**Comment by USB-05** [comment]
trigger gitleaks

**Comment by USB-05** [comment]
trigger gitleaks

**Comment by USB-05** [comment]
trigger gitleaks

**Review by raghavharness — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42944#-:-NMYOJFL3OteB6_Qj1Zz:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**